### DISCUSSION TITLE
[android]  chat dialog cannot be closed

### DISCUSSION THREAD

**Discussion by henkdegroot** [jamulussoftware/jamulus#1765] [UI Related]
**Describe the bug**
When the chat window opens on Android, it uses the fullscreen and does not allow to return (close) the window to get back to the main dialog.

**To Reproduce**
Open the app on Android and open the Chat window
When using a server that has "license" enabled, you automatically get the chat window displayed and cannot even the app at all.

**Expected behavior**
A close option should be available to the Chat window on Android device (perhaps on iOS as well).
The default x (available in other platforms) does not show on Android.

**Operating system**
Android 10

**Version of Jamulus**
3.8.0rc1

**Comment by ann0see** [comment]
I think https://github.com/jamulussoftware/jamulus/pull/1512 will fix this

**Comment by ngocdh** [comment]
Yes indeed, see https://github.com/jamulussoftware/jamulus/pull/1512/files#diff-0eda22ffb00bf0bbb2947f9e26d668a276a8934b68f231700d231676181ee5e8

**Comment by henkdegroot** [comment]
Wow...that is a very quick solution! Works correct indeed....I had seen the PR a while ago...and thought it was merged already.
Guess I was mistaken and should have checked first.

**Comment by dcorson-ticino-com** [comment]
I think the solution is a one window interface, if we decide to use it.
Here is a first stab at something useful: 
https://github.com/jamulussoftware/jamulus/discussions/1343#discussion-3290638

**Comment by ngocdh** [comment]
> one window interface

@dcorson-ticino-com I love it actually. But I have to disagree here, because of mobile devices' small screens. 
For desktop, I vote for one window interface.

**Comment by ann0see** [comment]
Yep. That's what Jamulus might look like in future ;-). However I suggested small steps to not confuse users.

**Comment by henkdegroot** [comment]
![mobile-preview](https://user-images.githubusercontent.com/13550012/119722883-1e0fd980-be6d-11eb-8a37-8f9c48203ae4.jpg)

It is not that bad at all on my mobile (screen resolution of 720 x 1480).....a few tweaks here and there...and we are back into the future :)

**Comment by dcorson-ticino-com** [comment]
> But I have to disagree here, because of mobile devices' small screens.

Then I am totally confused.  If the screen is too small, use a device with a screen size that is OK.
Why do you want to use it at all?  
I can't imagine how this should work if every time a chat comes the screen changes, one would spend more time playing with ones device than playing ones instrument.
The other guy who posted today about his Android had a screen resolution that is higher than on my PC.  With that even the normal Jamulus interface should work just as well as on a PC.
Apparently these devices don't let you use that resolution. :(

**Comment by ngocdh** [comment]
> It is not that bad at all on my mobile (screen resolution of 720 x 1480).....a few tweaks here and there...and we are back into the future :)

I'd love for those tweaks to happen. Because as it is now, if you're in a room with 10, you'll have to deal with small scrollbars which are hard to use (touch screen gestures not supported now). And those settings tabs, also hard to click on.

So if there's a way for everything to fit nicely and not so hard to manœuvre, I'm in.

**Comment by ngocdh** [comment]
> Why do you want to use it at all?  

Because it's practical.


> one would spend more time playing with ones device than playing ones instrument.

Which is why mobile apps must have their own design so that it's easy for users to use. Take chat window example: for mobile devices we could use notifications instead of the whole window.

Some mobile devices might have high resolution, but that also means controls are small on a (relatively) small screen, which makes them hard to use. At least that's what I get from using current Jamulus on mobile devices. 

But I'm sure we can make it better :)

**Reply by dcorson-ticino-com** [reply]
>Because it's practical.

Personally I don't see anything practical about it, only inconvenient and hard to use.

**Reply by ngocdh** [reply]
Sometimes, you might want to just take out your phone and listen to people jam. And how about a light setup allowing you to jam outdoor with friends?
Maybe you don't need all that. I personally do and would love to see the app improved on mobile :)

**Reply by dcorson-ticino-com** [reply]
Great! 
Then perfect your ideas and get down to programming.
How do you want to change the UI for a mobile version?

**Reply by ngocdh** [reply]
If only I had the skills. I think we need people with real mobile UI/UX experience.

**Reply by henkdegroot** [reply]
All starts with an idea first and then working on implementing and how that is done is an oppertunity.

Like I would see the single "view" GUI design (proposed by @dcorson-ticino-com) workable and perhaps addind swipe left/right options to it for showing the "extra" tabs. Say, I swipe from the left, the dialog with chat appears covering a x percent of the screen. Just do a quick chat and it disappaears automatic. I swipe from the right, perhaps show the Connectiong dialog and follow a similar principal.

A workflow of desirable actions and how to access them would be a good start. Perhaps some setting (like "My Profile") do not have a "swipe" dialog as usually it is a one of setting with every now and then an update. So good be done using a option in the menu.

At some point perhaps a decision needs to be made if the "mobile" needs it own branch, things like above can be more mobile specific.

Not sure if I would be a big fan of people joining my jam session using a phone...my current experience is that when everyone is using a network cable...all works fine. If there is one using WiFi everyone suffers during the session, regardless if they play or not.
Maybe that would be another suggestion, a real "listen only" mode.....just receiving the server audio stream an no audio upstream at all.

**Reply by ngocdh** [reply]
My idea: #1766 


>If there is one using WiFi everyone suffers during the session, regardless if they play or not.

You just have to mute the person causing trouble. If I'm not mistaken, Jamulus protocol doesn't allow one bad connection to drag the whole gang down.

**Reply by henkdegroot** [reply]
> You just have to mute the person causing trouble.

Thanks, it is still confusing for me why this happens....if the person joins with "mute myself" enabled, should I still "mute" that person?
As we still have this issue often during band rehearsals. If that does not make a difference, it might be an "active" player is still connected using WiFi and we hear that persons sounds make the bubble noises.

**Reply by ngocdh** [reply]
> if the person joins with "mute myself" enabled, should I still "mute" that person?

I don't think so, unless there's a bug in Jamulus which cancels "Mute myself".

> it might be an "active" player is still connected using WiFi and we hear that persons sounds make the bubble noises.

Yes one player can ruin it for everyone, until muted.

**Reply by dcorson-ticino-com** [reply]
> it might be an "active" player is still connected using WiFi and we hear that persons sounds make the bubble noises.

It is always difficult to know exactly what is happening through such a verbal description of the faulty sound, but in my experience wifi does more than bubble.  It is often causing short breaks in the sound.  For me "bubbling" means a poor connection from the ISP.  It happens when the packets arrive out-of-order.  As Jamulus already does packet reordering to some extent now since several versions we are not hearing about that as often as earlier.

**Comment by dcorson-ticino-com** [comment]
I moved this to discussion as we are far from resolution and still need to decide our direction here.
Much discussion is still necessary how to best continue.
